src/Qir/Controller/Executable/QirExecutableGenerator.cs

```diff
@@ -18,10 +24,44 @@ public QirExecutableGenerator(IClangClient clangClient, ILogger logger)
             this.logger = logger;
         }
 
-        public Task GenerateExecutableAsync(FileInfo executableFile, DirectoryInfo sourceDirectory, DirectoryInfo libraryDirectory, DirectoryInfo includeDirectory)
+        public async Task GenerateExecutableAsync(FileInfo executableFile, DirectoryInfo sourceDirectory, DirectoryInfo libraryDirectory, DirectoryInfo includeDirectory)
         {
-            // TODO: Compile and link libraries- "Microsoft.Quantum.Qir.Runtime", "Microsoft.Quantum.Qir.QSharp.Foundation", "Microsoft.Quantum.Qir.QSharp.Core"
-            throw new System.NotImplementedException();
+            // Wrap in a Task.Run because FileInfo methods are not asynchronous.
+            await Task.Run(async () =>
+            {
+                var binDirectory = executableFile.Directory;
+                logger.LogInfo($"Creating binary directory at {binDirectory.FullName}.");
+                executableFile.Directory.Create();
+
+                // Copy all library contents to bin.
+                logger.LogInfo("Copying library directory contents into the executable's folder.");
+                var libraryFiles = libraryDirectory.GetFiles();
+                foreach (var file in libraryFiles)
+                {
+                    CopyFileIfNotExists(file, binDirectory);
+                }
+
+                // Copy all include contents to bin.
+                logger.LogInfo("Copying include directory contents into the executable's folder.");
+                var includeFiles = includeDirectory.GetFiles();
+                foreach (var file in includeFiles)
+                {
+                    CopyFileIfNotExists(file, binDirectory);
```

Comment:
Include files should be copied to the `sourceDirectory` instead of the `binDirectory`.

Reply:
To not make clang compile headers, this should be done after determining the files that clang actually has to compile.

Reply:
Fixed, but I had to filter by extension. (Windows doesn't always let the program delete files if they are already present, and thus we have to assume that the files might have already been copied. I don't love it, but I suggest we don't spend time investigating that for now.)

Reply:
Reviewing the code again, include files don't need to be copied at all because the path to the folder that contains them is passed to clang through the `-I` option.

Reply:
Ah okay that's what I thought at first but I thought I was maybe missing something. I'll remove that!

Reply:
Removed it and it's still working!